airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/SyncWorkflowImpl.java

```diff
@@ -40,6 +53,25 @@ public StandardSyncOutput run(final JobRunConfig jobRunConfig,
                                 final StandardSyncInput syncInput,
                                 final UUID connectionId) {
 
+    final StandardCheckConnectionInput sourceConfiguration = new StandardCheckConnectionInput()
+        .withConnectionConfiguration(syncInput.getSourceConfiguration());
+    final StandardCheckConnectionInput destinationConfiguration =
+        new StandardCheckConnectionInput().withConnectionConfiguration(syncInput.getDestinationConfiguration());
+
+    final StandardCheckConnectionOutput sourceCheckResponse = checkActivity.check(jobRunConfig, sourceLauncherConfig, sourceConfiguration);
+    if (sourceCheckResponse.getStatus() == Status.FAILED) {
+      final Exception ex = new IllegalArgumentException(sourceCheckResponse.toString());
+      final FailureReason checkFailureReason = FailureHelper.checkFailure(ex, Long.valueOf(jobRunConfig.getJobId()), Math.toIntExact(jobRunConfig.getAttemptId()), FailureReason.FailureOrigin.SOURCE);
```

Comment:
Needing to convert `Long.valueOf(jobRunConfig.getJobId())` and ` Math.toIntExact(jobRunConfig.getAttemptId())` doesn't seem right... Any tips on where to get these values in the correct types? I *think* they really always are integers (workspaces, connectors, and connections are the UUIDs)